packages/actor-query-result-serialize-sparql-xml/lib/ActorQueryResultSerializeSparqlXml.ts

```diff
@@ -89,20 +88,22 @@ export class ActorQueryResultSerializeSparqlXml extends ActorQueryResultSerializ
       });
       resultStream.on('data', (bindings: Bindings) => {
         // XML SPARQL results spec does not allow unbound variables and blank node bindings
-        results.push({ result: [ ...bindings ]
-          .map(([ key, value ]) => ActorQueryResultSerializeSparqlXml.bindingToXmlBindings(value, key)) });
+        serializer.add({ name: 'result',
+          children: [ ...bindings ]
+            .map(([ key, value ]) => ActorQueryResultSerializeSparqlXml.bindingToXmlBindings(value, key)) });
       });
 
       // Close streams
       resultStream.on('end', () => {
-        results.close();
-        root.close();
+        serializer.close();
+        serializer.close();
```

Comment:
Should only be called once?

Reply:
No because we want to close both the `<results>` node and the `<sparql>` node.

Reply:
Ok, makes sense, thanks for clarifying!